### PR TITLE
Add streamlined ListItemGroup component

### PR DESCRIPTION
**Pull request by judastheo**
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

### CODE REVIEW

**Review by hcourthias — APPROVED** [review]

File: packages/sandbox/e2e/android/screenshots/ListItemGroup.png
Comment (kimak) [inline packages/sandbox/e2e/android/screenshots/ListItemGroup.png:1]: @ianrobertlambert to review on Android

File: packages/sandbox/e2e/ios/screenshots/ListItemGroup.png
Comment (kimak) [inline packages/sandbox/e2e/ios/screenshots/ListItemGroup.png:1]: @ianrobertlambert to review on iOS

**Comment by kimak** [comment]
Nice! We will be able to update the Sandbox with it 🤩

**Review by kimak — APPROVED** [review]

**Comment by ianrobertlambert** [comment]
Nice @judastheo! 🙌 Looks good from my point of view. Looking forward to seeing these on the sandbox! 😊 Super small feedback:

- [x] Can we enforce a rule so that the last `ListItem` never has a divider?

_(In the sandbox, there is a `ListItemGroup` with one `ListItem` and it has a divider.)_

I had one other small comment on the touch interaction color for the `ListItem`, but I will create a separate issue for that.

File: packages/sandbox/e2e/android/screenshots/ListItemGroup.png
Comment (ianrobertlambert) [reply packages/sandbox/e2e/android/screenshots/ListItemGroup.png:1]: Validated 👌 Thanks!

File: packages/sandbox/e2e/ios/screenshots/ListItemGroup.png
Comment (ianrobertlambert) [reply packages/sandbox/e2e/ios/screenshots/ListItemGroup.png:1]: Validated 👌

**Comment by judastheo** [comment]
> Nice @judastheo! 🙌 Looks good from my point of view. Looking forward to seeing these on the sandbox! 😊 Super small feedback:
> 
> * [ ]  Can we enforce a rule so that the last `ListItem` never has a divider?
> 
> _(In the sandbox, there is a `ListItemGroup` with one `ListItem` and it has a divider.)_
> 
> I had one other small comment on the touch interaction color for the `ListItem`, but I will create a separate issue for that.

Nice catch @ianrobertlambert , the rule I implemented was off, fix on the way!

**Pull request merged**